icu4c/source/i18n/rematch.cpp

```diff
@@ -717,7 +717,6 @@ UBool RegexMatcher::find(UErrorCode &status) {
             if  (findProgressInterrupt(startPos, status))
                 return FALSE;
         }
-        U_ASSERT(FALSE);
 
```

Comment:
Agree.  This question has been brought up before but I don't think much came of it.  Can we add it right now?  Just go add a macro called `U_UNREACHABLE` or `UPRV_UNREACHABLE` in a file like platform.h.

Reply:
Thanks Shane. I wonder if it might be better to have it be private (aka `UPRV_`), at least at first?

Reply:
Thanks for the feedback! I'm going to file another ticket for this because this PR is quite large as it is. There also might be implications from adding a new private macro to the public header platform.h.